From a15507017d61a512b69eafe087df321ff02e4981 Mon Sep 17 00:00:00 2001
From: Alexander Grund <alexander.grund@tu-dresden.de>
Date: Mon, 12 Oct 2020 13:39:21 +0200
Subject: [PATCH] Undefine bool, vector, pixel after including altivec.h

This avoids failures when using those keywords later on in C++ code.
---
 aten/src/ATen/cpu/vec256/intrinsics.h | 3 +++
 aten/src/ATen/native/cpu/Intrinsics.h | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/aten/src/ATen/cpu/vec256/intrinsics.h b/aten/src/ATen/cpu/vec256/intrinsics.h
index fc3fd8547d81c..aa94fd8ba3bb5 100644
--- a/aten/src/ATen/cpu/vec256/intrinsics.h
+++ b/aten/src/ATen/cpu/vec256/intrinsics.h
@@ -32,6 +32,9 @@
         (defined(__VEC__) || defined(__ALTIVEC__))
 /* XLC or GCC-compatible compiler, targeting PowerPC with VMX/VSX */
 #include <altivec.h>
+#undef bool
+#undef vector
+#undef pixel
 #elif defined(__GNUC__) && defined(__SPE__)
 /* GCC-compatible compiler, targeting PowerPC with SPE */
 #include <spe.h>
diff --git a/aten/src/ATen/native/cpu/Intrinsics.h b/aten/src/ATen/native/cpu/Intrinsics.h
index c4c1d6b648a37..86e1d5005b6e4 100644
--- a/aten/src/ATen/native/cpu/Intrinsics.h
+++ b/aten/src/ATen/native/cpu/Intrinsics.h
@@ -22,6 +22,9 @@
     (defined(__VEC__) || defined(__ALTIVEC__))
 /* XLC or GCC-compatible compiler, targeting PowerPC with VMX/VSX */
 #include <altivec.h>
+#undef bool
+#undef vector
+#undef pixel
 #elif defined(__GNUC__) && defined(__SPE__)
 /* GCC-compatible compiler, targeting PowerPC with SPE */
 #include <spe.h>
